src/components/Collapse.tsx

```diff
@@ -24,6 +24,7 @@ export const Collapse = ({ className, isOpen, label, children, onClick, ...props
           onClick?.(!isOpenState);
         }}
         type="button"
+        data-fs-element={`Collapse header ${label}`}
```

Comment:
This is a React.Node so you can't really use this one. Might be worth adding it higher up and letting this accept `...rest` props so it is added automatically. (Meant to leave this comment with the review -- also shows how I discovered it quickly as it logged [Object object]).

Reply:
Hmmm yeah. I've updated it to come from above, but since we're already using the `...rest` pattern here with a destination that isn't the button, the syntax gets a little gross. Sort of hinges on how accurate the `document.addEventListener...` simulation is, because putting the attribute on the wrapping div might or might not do the thing we want. I can't find anything more specific in the docs than "We look at the DOM"

Reply:
>  Sort of hinges on how accurate the `document.addEventListener...` simulation is, because putting the attribute on the wrapping div might or might not do the thing we want. I can't find anything more specific in the docs than "We look at the DOM"

I rewrote the simulation slightly so it recursively checks parent elements for the `data-fs-element` attribute because I figure they must be doing something like that in the worst case:

```
document.addEventListener(
  'focus',
  (e) => {
    if (e.target instanceof HTMLElement) {
      logDomChain({ node: e.target });
    }
  },
  true
);

type logDomChainProps = {
  parentChain?: string[];
  node: HTMLElement | null;
};

function logDomChain({ parentChain = [], node }: logDomChainProps) {
  if (node === null) {
    return console.log(constructChain(parentChain));
  }

  const dataFsElement = node.getAttribute(`data-fs-element`);

  if (dataFsElement) {
    return console.log(constructChain([...parentChain, dataFsElement]));
  }

  const newChain = [...parentChain, node.tagName.toLowerCase()];
  return logDomChain({ node: node.parentElement, parentChain: newChain });
}

function constructChain(chain: string[]) {
  return [...chain].reverse().join(` > `);
}
```

This is showing me that none of the Selects or Sliders are receiving the attribute and is also showing me some other fields we haven't tagged up. (I figure we might as well try and remove as much noise as possible, right?).

In those cases where we can't inject it we could either write some 'fun-but-not-really'™️ React code to target the underlying elements or just try adding it to the wrapping `<Field />` component or such?

![A screenshot of Chrome's console dev tools with a list of selectors. Some of them are showing the values for data-fs-element whilst others are showing a very long html tag chain up to the document body.](https://github.com/grafana/synthetic-monitoring-app/assets/6906380/6a15ad7e-fb9d-427f-a07d-43eebe2c2287)

Reply:
Nice! Looking at the fullstory results, it looks like it will report the `id` in a readable way if it's present. Maybe that's a way out for select/slider that won't let us pass arbitrary attributes down to the dom?